helix-term/src/ui/completion.rs

```diff
@@ -97,6 +97,16 @@ impl Completion {
         start_offset: usize,
         trigger_offset: usize,
     ) -> Self {
+        // Sort completion items according to their preselect status (given by the LSP server)
+        let mut items = items;
+        items.sort_by(|item1, item2| {
+            item2
+                .preselect
+                .unwrap_or(false)
+                .cmp(&item1.preselect.unwrap_or(false))
+        });
```

Comment:
Should be possible to use `sort_by_key` (or better yet `sort_unstable_by_key`)

```rust
items.sort_unstable_by_key(|item| item.preselect.unwrap_or(false) )
```

Reply:
Done, although we need to reverse the conditions that that the ones that have `preselect == Some(true)` are first in the completion menu.
Using `sort_unstable_by_key` would lose the order given by the LSP server, which, with what I've tested, does have its significance.